rllib/tests/test_model_imports.py

```diff
@@ -0,0 +1,210 @@
+        else:
+            model_out = self.base_model(input_dict["obs"])
+            self._value_out = tf.zeros(
+                shape=(tf.shape(input_dict["obs"])[0], ))
+        return model_out, state
+
+    def value_function(self):
+        return tf.reshape(self._value_out, [-1])
+
+    def import_from_h5(self, import_file):
+        # Override this to define custom weight loading behavior from h5 files.
+        self.base_model.load_weights(import_file)
+
+
+class MyTorchModel(TorchModelV2, nn.Module):
+    """Generic vision network."""
+
+    def __init__(self, obs_space, action_space, num_outputs, model_config,
+                 name):
+        TorchModelV2.__init__(self, obs_space, action_space, num_outputs,
+                              model_config, name)
+        nn.Module.__init__(self)
+
+        self.device = torch.device("cuda"
+                                   if torch.cuda.is_available() else "cpu")
+
+        self.layer_1 = nn.Linear(obs_space.shape[0], 16).to(self.device)
+        self.layer_out = nn.Linear(16, num_outputs).to(self.device)
+        self.value_branch = nn.Linear(16, 1).to(self.device)
+        self.cur_value = None
+
+    def forward(self, input_dict, state, seq_lens):
+        layer_1_out = self.layer_1(input_dict["obs"])
+        logits = self.layer_out(layer_1_out)
+        self.cur_value = self.value_branch(layer_1_out).squeeze(1)
+        return logits, state
+
+    def value_function(self):
+        assert self.cur_value is not None, "Must call `forward()` first!"
+        return self.cur_value
+
+    def import_from_h5(self, import_file):
+        # Override this to define custom weight loading behavior from h5 files.
+        f = h5py.File(import_file)
+        self.layer_1.load_state_dict({
+            "weight": torch.Tensor(
+                np.transpose(f["layer1"]["default_policy"]["layer1"][
+                    "kernel:0"].value)),
+            "bias": torch.Tensor(
+                np.transpose(
+                    f["layer1"]["default_policy"]["layer1"]["bias:0"].value)),
+        })
+        self.layer_out.load_state_dict({
+            "weight": torch.Tensor(
+                np.transpose(
+                    f["out"]["default_policy"]["out"]["kernel:0"].value)),
+            "bias": torch.Tensor(
+                np.transpose(
+                    f["out"]["default_policy"]["out"]["bias:0"].value)),
+        })
+        self.value_branch.load_state_dict({
+            "weight": torch.Tensor(
+                np.transpose(
+                    f["value"]["default_policy"]["value"]["kernel:0"].value)),
+            "bias": torch.Tensor(
+                np.transpose(
+                    f["value"]["default_policy"]["value"]["bias:0"].value)),
+        })
+
+
+def model_import_test(algo, config, env):
+    # Get the abs-path to use (bazel-friendly).
+    rllib_dir = Path(__file__).parent.parent
+    import_file = str(rllib_dir) + "/tests/data/model_weights/weights.h5"
```

Comment:
Instead of checking this file in, could we generate it on the fly by calling policy.model.save()? I think this would help users understand how to save too.

Reply:
Yeah, but we don't have any means to currently store an h5 file.
I think we should take a step back and write a design doc first to make sure we create a clean save/restore API (preferably controllable from the top trainer level, I think).
This PR was really just to close the issue and make sure this ad-hoc import from an h5 file will work at all.